src/main/java/com/atlassian/db/replica/spi/DatabaseCluster.java

```diff
@@ -0,0 +1,10 @@
+package com.atlassian.db.replica.spi;
+
+import com.atlassian.db.replica.api.Database;
+
+import java.sql.SQLException;
+import java.util.stream.Stream;
+
+public interface DatabaseCluster {
+    Stream<Database> getReplicas() throws SQLException;
```

Comment:
I think we'd rather have a collection here

Reply:
Why?

Reply:
Streams can be potentially infinite, you cannot use them more than once

Reply:
+1 for Collection.
Why Streams?

Reply:
Streams or Iterables give more flexibility to implementations. I thought about future implementations that could create connections lazily, but I think `Supplier<Connection>` will work even better. For aurora we know we can get data with a single call, so no benefit from lazily initialized Database stream. Will it be true for other database clusters? Will anyone need it? I don't know.

Usually, it's safer to start with a stream/iterable API when the collection can be large. This way you can fix performance problems without breaking the API. I think this time we can use Collection and see how it evolves.

Reply:
> can be potentially infinite,

The consumer controls the flow, so I see no problem with that.


>  you cannot use them more than once

It's something everyone should be aware of. In case you need to use it multiple times, you store it in a collection.

Reply:
I'm fine with Collection 👌